server.js

```diff
@@ -20,4 +24,81 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+
+//  Create a new message
+app.use(express.json());
+
+app.post("/messages", function (req, res) {
+  const newMsg = req.body;
+  console.log(req);
+  
+  // Level 2 - simple validation
+  const msg = {
+    id: idCount++,
+    from: req.body.from,
+    text: req.body.text,
+  };
+  if (!newMsg.from || !newMsg.text) {
+    res.status(400).send("Rejected: empty or missing text or from property.");
+  } else {
+    messages.push(msg);
+    // res.status(201).send(`Request accepted`);
+    res.redirect("/");
+  }
+});
+
+
+//  Read all messages
+app.get("/messages", function (req, res) {
+  res.status(200).send({ messages });
+});
+
+
+// Read only the most recent 10 messages: /messages/latest
+app.get("/messages/latest", function (req, res) {
+  if (messages.length <= 10) {
+    res.status(200).send({ messages });
+  } else {
+    const latest10 = messages.slice(-10);
+    res.status(200).send({ latest10 });
+  }
+});
+
+
+// Read only messages whose text contains a given substring: /messages/search?text=express
+app.get("/messages/search", function (req, res) {
+  const searchWord = req.query.text;
+  const result = search(searchWord);
+  res.json(result);
+});
+
+//search by a term
+function search(word) {
+  return messages.filter(
+    (msg) => msg.text.toLowerCase().includes(word.toLowerCase())
+  );
+}
+
+
+//  Read one message specified by an ID
+app.get("/messages/:msgId", function (req, res) {
+  const idToFind = Number(req.params.msgId);
+  const msg = messages.find((msg) => msg.id === idToFind);
+  res.status(200).send({ msg });
+});
```

Comment:
this sends back the requested message, nice job (also not sure if intended but at the moment, the message is returned with the key of `msg`). Have a think what you should send back if the requested id doesn't exist